v3/pool_x.go

```diff
@@ -0,0 +1,42 @@
+// +build linux darwin freebsd netbsd openbsd solaris dragonfly plan9 aix
+
+package pb
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cbehopkins/pb/v3/termutil"
+)
+
+func (p *Pool) print(first bool) bool {
+	p.m.Lock()
+	defer p.m.Unlock()
+	var out string
+	if !first {
+		out = fmt.Sprintf("\033[%dA", p.lastBarsCount)
+	}
+	isFinished := true
+	bars := p.bars
+	rows, _, _ := termutil.TerminalSize()
+	// if err != nil {
+	// 	cols = defaultBarWidth
+	// }
+	if rows > 0 && len(bars) > rows {
+		bars = bars[:rows]
+	}
+	for _, bar := range bars {
+		if !bar.IsFinished() {
+			isFinished = false
+		}
+		// bar.SetWidth(cols)
```

Comment:
As I wrote above... why we can't just call `TerminalSize` once and set the width here to avoid extra syscalls?

Reply:
So this was to fix an inconsistency between pool_x and pool_win.
Sorry should have deleted it rather than commented.

By setting the width here, you are setting it to the default, not to the current. (just above you have read it from the TerminalSize, overwriting any setting made in the pb itself.

Or am I misunderstanding the intent here?

Reply:
This is done on purpose to call TerminalSize once per print and not for each bar. I have not changed the windows version because not able to test it.

Reply:
So the intent is to remove the ability for the user to set column width of a bar when running in pool mode?

Reply:
What are you think about to add SetWidth method to the pool?

Reply:
I like that

Reply:
I've pushed something to my fork